pkg/deploy/deploy.go

```diff
@@ -158,20 +159,42 @@ func (d *deployer) deploy(ctx context.Context, rgName, deploymentName, vmssName
 	for i := 0; i < 3; i++ {
 		d.log.Printf("deploying %s", deploymentName)
 		err = d.deployments.CreateOrUpdateAndWait(ctx, rgName, deploymentName, deployment)
-		if serviceErr, ok := err.(*azure.ServiceError); ok &&
-			serviceErr.Code == "DeploymentFailed" &&
-			i < 1 {
-			// on new RP deployments, we get a spurious DeploymentFailed error
-			// from the Microsoft.Insights/metricAlerts resources indicating
-			// that rp-lb can't be found, even though it exists and the
-			// resources correctly have a dependsOn stanza referring to it.
-			// Retry once.
-			d.log.Print(err)
-			continue
+		serviceErr, isServiceError := err.(*azure.ServiceError)
+
+		// Check for two known errors that we know how to handle.
+		if isServiceError && serviceErr.Code == "DeploymentFailed" && len(serviceErr.Details) > 0 {
+			outerCode, outerCodeOk := serviceErr.Details[0]["code"].(string)
+			message, messageOk := serviceErr.Details[0]["message"].(string)
+
+			if outerCodeOk && messageOk && outerCode == "BadRequest" {
+				innerErr := map[string]interface{}{}
```

Comment:
When you say to use a struct do you mean to unmarshal the inner error into an `azure.ServiceError` instead of a `map`?

If that's what you meant, I was trying to do that at first and it didn't work for some reason, but I'll go back and take one more look. It would definitely be nicer than having to do the type casting.

Reply:
you can have a custom struct that you unmarshal into. here for example you can get your innerErr unmarshalled into something like this:
```go
type innerError struct{
    code string
    message string
}
```
and then you'd avoid the type casting on line 173/174

Reply:
I'm not sure what I was doing wrong when I first tried, but casting the nested errors to the `ServiceError` type ended up working fine (and I've committed the change).